### DISCUSSION TITLE
Editing NPM dist files

### DISCUSSION THREAD

**Discussion by bhison** [uber/react-map-gl#1471] [Q&A]
Currently using this awesome library in an NPM project but have decided I'd like to extend the customisability of NavigationControl so started looking at adding some of my own updates. This isn't something I have done elsewhere before so I apologise if I'm particularly ignorant with my approach here, all advice appreciated.

I had hoped to use [patch-package](https://www.npmjs.com/package/patch-package), at least to initially dip my toes in, but I've noticed that the dist folder of react-map-gl's NPM module is seemingly pre-transpiled from jsx to js syntax which for me at least makes things a bit harder. Is there a script or process I can use to make edits to the src version of a file and update it's counterpart in the dist folder? Thanks all in advance and apologies again if this is something very fundemental I am yet to have learned.

**Comment by bhison** [comment]
I realise this is probably more to do with my understanding of NPM than anything else. If admin would like to delete, that would be fine with me. Thanks again.

**Comment by Pessimistress** [comment]
You can set your bundler to resolve `react-map-gl` to `@node_modules/react-map-gl/src`. E.g. for webpack use [resolve.alias](https://webpack.js.org/configuration/resolve/#resolvealias)
